### ISSUE TITLE
"Deleting a list item while iterating over it" has improper example or explanation

### ISSUE THREAD

**Issue by nolar** [satwikkansal/wtfpython#16]
In the "Deleting a list item while iterating over it" trick, in the `list_1` example, `del` actually deletes the `item` variable name. The name will be reassigned to the next value on the next iteration, but will be absent up to the end of the code block after the `del item` statement. No operator even tries to delete from the `list_1`, and the list remains unaffected.

There are 2 possible ways to fix this:

1. Either you meant `del list_1[idx]` in the example, which removes the indexed item from the list. And the output will be `[2, 4]` for the same reason as with `list_4.pop(idx)`.
2. Or the explanation is wrong, and should not refer to item deletion, but should refer to the variable name un-assignment.

Just for historic reference, this is how the example looks now:

```
list_1 = [1, 2, 3, 4]
for idx, item in enumerate(list_1):
    del item
```

```
>>> list_1
[1, 2, 3, 4]
```

```
* del removes a specific index (That's why first list_1 was unaffected), raises IndexError if an invalid index is specified.
```

**Comment by satwikkansal** [comment]
Ah, right, my only intention to add `del` was to show the abuse of the `del` keyword. I'll fix the explanation.

**Comment by satwikkansal** [comment]
@nolar Thanks for reporting the issue! I've fixed this in 7e93e0c, feel free to reopen the issue if the explanation seems insufficient or incorrect :+1:
